core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -337,6 +337,17 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
       _ <- gcsDAO.grantReadAccess(billingProjectName, xfer.bucket, Set(ownerGroupEmail, computeUserGroupEmail))
     } yield {
       RequestComplete(StatusCodes.Created)
+    }).recoverWith {
+      case t: Throwable =>
+        // attempt cleanup then rethrow
+        for {
+          _ <- samDAO.deleteResource(SamResourceTypeNames.billingProject, project.projectName.value, ownerUserInfo).recover {
+            case x => logger.debug(s"failure deleting billing project ${project.projectName.value} from sam during error recovery cleanup.", x)
```

Comment:
shall we log at `info` or `warn`?

Reply:
we are already recovering from a failure which will get logged at a higher level... I expect 1 or both of these to fail so debug seems right